contracts/Comet.sol

```diff
@@ -765,22 +765,10 @@ contract Comet is CometMainInterface {
      */
     function doTransferIn(address asset, address from, uint amount) internal returns (uint) {
         uint256 preTransferBalance = IERC20NonStandard(asset).balanceOf(address(this));
-        IERC20NonStandard(asset).transferFrom(from, address(this), amount);
-        bool success;
-        assembly ("memory-safe") {
-            switch returndatasize()
-                case 0 {                       // This is a non-standard ERC-20
-                    success := not(0)          // set success to true
-                }
-                case 32 {                      // This is a compliant ERC-20
-                    returndatacopy(0, 0, 32)
-                    success := mload(0)        // Set `success = returndata` of override external call
-                }
-                default {                      // This is an excessively non-compliant ERC-20, revert.
-                    revert(0, 0)
-                }
+        (bool success, bytes memory returndata) = asset.call(abi.encodeWithSelector(IERC20NonStandard.transferFrom.selector, from, address(this), amount));
+        if (!success || !(returndata.length == 0 || abi.decode(returndata, (bool)))) {
```

Comment:
I think `(returndata.length != 0 && !abi.decode(returndata, (bool)))` is easier to understand than `!(returndata.length == 0 || abi.decode(returndata, (bool)))`. Same for `doTransferOut`